### PR TITLE
added backend route for fetching topic data

### PR DESCRIPTION
**Pull request by Icelain**
Route is: /api/get-topics. Partially improves #152

### CODE REVIEW

**Comment by Lalit3716** [comment]
The endpoint looks okay! but I was thinking if you could fetch the topics data from the MongoDB database?

**Comment by Icelain** [comment]
> The endpoint looks okay! but I was thinking if you could fetch the topics data from the MongoDB database?

Oh yeah I was wondering about that. Didn't know earlier but thanks for the confirmation. I'll work on it.

**Comment by Icelain** [comment]
hey I made the changes. Is it okay?

**Comment by Lalit3716** [comment]
Yeah, it looks good but it would be great if you could paginate the response. You can see this [controller](https://github.com/Vita-App/Vita/blob/main/api/src/controllers/api.controller.ts#L11) for reference.

File: api/src/controllers/topics.controller.ts [hunk 1]
```diff
@@ -2,9 +2,36 @@ import { Request, Response } from 'express';
...
 const getTopics = async (req: Request, res: Response) => {
-  const topicsData: Topic[] = await TopicModel.find({});
-  res.status(200).json(topicsData);
+  const page = Number(req.query.page || '1');
+
+  let totalPages = 0;
+  let nextPage: number | null = null;
+  let prevPage: number | null = null;
+
+  const limit = Number(req.query.limit) === 0 ? 15 : Number(req.query.limit);
```
Comment (Lalit3716) [inline api/src/controllers/topics.controller.ts:13]: Change default to 0 i.e. no limit

**Comment by Icelain** [comment]
I made the limit 0 by default :)

**Comment by Lalit3716** [comment]
LGTM 🎉

**Pull request merged**